src/EnergyPlus/DataHeatBalance.hh

```diff
@@ -2135,46 +2135,59 @@ namespace DataHeatBalance {
         }
     };
 
-    struct HeatReclaimRefrigeratedRackData
+    struct HeatReclaimDataBase
     {
         // Members
-        std::string Name;       // Name of refrigerated rack
-        std::string SourceType; // object type for refrigerated rack
+        std::string Name;       // Name of DX Coil
         Real64 AvailCapacity;   // Total available heat reclaim capacity
+
+        // Default Constructor
+        HeatReclaimDataBase() : AvailCapacity(0.0)
+        {
+        }
+    };
+
+    struct HeatReclaimRefrigeratedRackData: HeatReclaimDataBase        //inherented from base struct
+    {
+        // Customized Members
+        std::string SourceType; // SourceType for DX Coil
```

Comment:
``SourceType`` in ``HeatReclaimRefrigCondenserData`` struct is int while others are string. Looks a little bit weird the consistency. I am not quite following the refactor issue. What can I do to avoid mess?

Reply:
Ugh. I didn't notice one of them was integer. Actually, that's the only one that is ever used, and it's only used to generate an error message in HeatingCoils::GetHeatingCoilInput. More that could be simplified here. This PR follows the awkward pattern that was already in the code for desuperheating coils. I can't expect this PR to fix all of the old problems, but it would be nice to make a little progress and remove the duplicate index searching that's happening in `GetWaterThermalTankInputData` (see this [comment](https://github.com/NREL/EnergyPlus/pull/7251#discussion_r294995201))

Reply:
Haha understand. I'd love to clean the parts I touched. I am now working on the changes for that comment.